### PR TITLE
SDIT-800 Link up the activities api migrate appointments endpoint

### PR DESCRIPTION
**Pull request by steverendell**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/appointments/AppointmentResponse.kt [hunk 1]
```diff
@@ -27,4 +27,12 @@ data class AppointmentResponse(
...
   @Schema(description = "Status")
   val status: String,
+  @Schema(description = "Record created date", required = true)
+  val createdDate: LocalDateTime,
+  @Schema(description = "Record created by", required = true)
+  val createdBy: String,
+  @Schema(description = "Record modified date")
+  val modifiedDate: LocalDateTime? = null,
+  @Schema(description = "Record modified by")
+  val modifiedBy: String? = null,
```
Comment (steverendell) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/appointments/AppointmentResponse.kt:37]: these fields need migrating too

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/appointments/AppointmentService.kt [hunk 2]
```diff
@@ -172,6 +172,10 @@ private fun mapModel(entity: OffenderIndividualSchedule): AppointmentResponse =
     status = entity.eventStatus.code,
     subtype = entity.eventSubType.code,
     internalLocation = entity.internalLocation?.locationId,
-    prisonId = entity.prison?.id,
+    prisonId = entity.prison?.id ?: entity.toPrison?.id,
```
Comment (steverendell) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/appointments/AppointmentService.kt:175]: In a few cases in the existing data the prison is null but the 'to_prison' is populated

File: src/main/resources/db/migration/V1_65__OFFENDER_IND_SCHEDULES.sql [hunk 3]
```diff
@@ -175,3 +175,7 @@ CREATE INDEX "OFF_IND_SCH_AGY_LOC_FK2"    ON "OFFENDER_IND_SCHEDULES" ("TO_AGY_L
 CREATE INDEX "OFF_IND_SCH_OFF_MOV_APP_FK" ON "OFFENDER_IND_SCHEDULES" ("OFFENDER_MOVEMENT_APP_ID");
 CREATE INDEX "OFF_IND_SCH_OFF_PRG_OB_FK"  ON "OFFENDER_IND_SCHEDULES" ("OFFENDER_PRG_OBLIGATION_ID");
 CREATE INDEX "OFF_IND_SCH_STF_FK"         ON "OFFENDER_IND_SCHEDULES" ("IN_CHARGE_STAFF_ID");
+
+CREATE TRIGGER OFFENDER_IND_SCHEDULES_TA
+  BEFORE UPDATE on OFFENDER_IND_SCHEDULES FOR EACH ROW
+  CALL "uk.gov.justice.digital.hmpps.nomisprisonerapi.helper.ModifyTrigger";
```
Comment (steverendell) [inline src/main/resources/db/migration/V1_65__OFFENDER_IND_SCHEDULES.sql:181]: This mimics Oracle behaviour, populating the modified columns

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/OffenderIndividualSchedule.kt [hunk 4]
```diff
@@ -86,12 +90,32 @@ class OffenderIndividualSchedule(
...
   @Column(name = "COMMENT_TEXT")
   val comment: String? = null,
+
+  @Column(name = "CREATE_DATETIME", nullable = false, updatable = false, insertable = false)
+  @CreatedDate
+  val createdDate: LocalDateTime? = null,
+
+  @Column(name = "CREATE_USER_ID", nullable = false, updatable = false, insertable = false)
+  @CreatedBy
+  val createdBy: String? = null,
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/OffenderIndividualSchedule.kt:110]: It says nullable=false so type should match?

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**